### PR TITLE
Add support for django 1.10

### PR DESCRIPTION
**Pull request by mokutsu**
-Add try except statements around changed syntax in django 1.10
-modify middleware for compatibility
-add AUTH_HASH_SESSION_KEY for compatibility with django.contrib.auth.middleware.SessionAuthenticationMiddleware

### CODE REVIEW

File: django_switchuser/urls.py [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
-urlpatterns = patterns("",
+patterns = [
     url(r"^$", views.su_login, name="su-login"),
     url(r"^logout$", views.su_logout, name="su-logout"),
-)
+]
+
+try:
+    urlpatterns = patterns("", *patterns)
+except NameError:
```
Comment (wolever) [inline django_switchuser/urls.py]: 1. The `patterns = […]` clobbers the `from … import patterns`
2. Why would a `NameError` be raised? (this isn't something I've seen before)

Comment (mokutsu) [inline django_switchuser/urls.py]: You're totally right, I also missed the import for patterns as well.  If we try to use 'patterns' after the import fails, I think we end up with `NameError: name 'patterns' is not defined`. 


I've updated it just to explicitly check for the django version instead, hopefully that's a bit clearer!

**Comment by wolever** [comment]
Thanks! I'm going to merge this and test it, then cut a new release next week :)

**Comment by wolever** [comment]
Would you be able to pull and test the most recent version? It's got another bugfix that _should_ be just fine, but confirmation would be nice before I push!

**Comment by mokutsu** [comment]
@wolever pulled it down and works! thanks for cleaning up the changes!

**Pull request merged**